src/sprintf.js

```diff
@@ -131,7 +131,7 @@
                     }
                     pad_character = ph.pad_char ? ph.pad_char === '0' ? '0' : ph.pad_char.charAt(1) : ' '
                     pad_length = ph.width - (sign + arg).length
-                    pad = ph.width ? (pad_length > 0 ? pad_character.repeat(pad_length) : '') : ''
+                    pad = ph.width ? (pad_length > 0 ? new Array(pad_length + 1).join(pad_character) : '') : ''
```

Comment:
Funny thiats similar solution to one removed https://github.com/alexei/sprintf.js/pull/139/commits/4ed9a44203c3ce77ed8b763b27631ced882d4eea#diff-13ed28d46a5f76f4d44561850bda81bbL222